addons/website_event_booth_sale/models/sale_order.py

```diff
@@ -42,9 +42,36 @@ def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
                     event_booth_registrations_command = new_registrations_commands
                 values['event_booth_registration_ids'] = event_booth_registrations_command
 
+            discount = 0
+            order = self.env['sale.order'].sudo().browse(order_id)
+            booth_currency = booths[:1].product_id.currency_id
```

Comment:
If multiple booths can be specified here, I think you should make it work correctly even if one of the booth product is company restricted :P.